MonstieWash/Assets/Scripts/MonsterAi/MonsterBrain.cs

```diff
@@ -0,0 +1,308 @@
+    }
+
+    /// <summary>
+    /// Takes all moodtypes and stores them in activeMood dictionary, where first value is their index in moodtypes list and the second is their current value;
+    /// </summary>
+    private void loadMoods()
+    {
+        for(int i = 0; i < moodData.Count; i++)
+        {
+            activeMoods.Add(i, moodData[i].MoodStartingPoint);
+            activeMoodNames.Add(moodData[i].MoodName,i);
+        }
+    }
+
+
+    /// <summary>
+    /// Updates ui textbox with information useful to debugging.
+    /// </summary>
+    private void updateDebugText()
+    {
+        DebugUi.text = "";
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            DebugUi.text += "MoodName: " + moodData[i].MoodName + " MoodValue: " + Mathf.FloorToInt(activeMoods[i]).ToString() + "\n" + 
+            " Mood Lower/Upper Limits: " + moodData[i].MoodLowerLimit.ToString() + "/" + moodData[i].MoodUpperLimit.ToString();
+            DebugUi.text += "\n";
+            DebugUi.text += "\n";
+        }
+
+    }
+
+
+    /// <summary>
+    /// Returns the index of a moodtype within active moods by its string name.
+    /// </summary>
+    /// <param name="name"> The name of the desired mood index</param>
+    /// <returns></returns>
+    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
+    private int accessActiveMoodIndexByName(string name)
+    {
+        return activeMoodNames[name];
+    }
+
+    /// <summary>
+    /// Returns the index of a moodtype within active moods by a scriptable object reference.
+    /// </summary>
+    /// <param name="name"> The name of the desired mood index</param>
+    /// <returns></returns>
+    /// <exception cref="System.Exception"> When no mood exists with that name </exception>
+    private int accessActiveMoodIndexByMoodType(MoodType refMT)
+    {
+        return activeMoodNames[refMT.MoodName];
+    }
+
+
+    /// <summary>
+    /// Updates a mood with the given name by the amount given, useful for other scripts to interact with this.
+    /// </summary>
+    /// <param name="amount"> The float amount to change the mood by.</param>
+    /// <param name="name"> The name of the mood desired to change.</param>
+    public void updateMoodByName(float amount, string name)
+    {
+        var index = accessActiveMoodIndexByName(name);
+        activeMoods[index] += amount;
+        maintainLimit(index);
+    }
+
+    /// <summary>
+    /// Updates a mood with the given moodtype object by the amount given, useful for other scripts to interact with this.
+    /// </summary>
+    /// <param name="amount"></param>
+    /// <param name="mt"></param>
+    public void updateMoodByType(float amount, MoodType mt)
+    {
+        var index = accessActiveMoodIndexByMoodType(mt);
+        activeMoods[index] += amount;
+        maintainLimit(index);
+
+    }
```

Comment:
Same thing with these two, they can be overloaded.
```suggestion
    /// <summary>
    /// Updates a mood with the given name by the amount given, useful for other scripts to interact with this.
    /// </summary>
    /// <param name="amount"> The float amount to change the mood by.</param>
    /// <param name="name"> The name of the mood desired to change.</param>
    public void updateMood(float amount, string name)
    {
        var index = accessActiveMoodIndexByName(name);
        activeMoods[index] += amount;
        maintainLimit(index);
    }

    /// <summary>
    /// Updates a mood with the given moodtype object by the amount given, useful for other scripts to interact with this.
    /// </summary>
    /// <param name="amount"></param>
    /// <param name="mt"></param>
    public void updateMood(float amount, MoodType mt)
    {
        var index = accessActiveMoodIndexByMoodType(mt);
        activeMoods[index] += amount;
        maintainLimit(index);

    }
```